lib/adapter.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const Pensuer = require('penseur');
+const Hoek = require('hoek');
+const Path = require('path');
+
+const Adapter = function (params) {
+
+    this.params = params || {};
+    if (!this.params.database) {
+        throw new Error('Database Connection params should be set');
+    }
+};
+
+Adapter.prototype.getTemplatePath = function () {
+
+    return Path.join(__dirname, 'migrationTemplate.js');
+};
+
+Adapter.prototype.connect = function (callback) {
+
+    const self = this;
+    const settings = Hoek.clone(self.params.database.connection);
+    this.db = new Pensuer.Db(self.params.database.name, settings);
+    //ensure migrations table exists in the database
+    if (self.params.tables.indexOf('migration') === -1) {
```

Comment:
It would be nice to be able to pass in the name of the table that you want to use to store the migration, but it would have to be used in all places where the migrations table is references (not sure if that was the intention).